design-docs/Operation execution.md

```diff
@@ -0,0 +1,78 @@
+# Operation execution
+
+An overview of the main classes and process involved when executing an operation (queries and mutation only -
+subscriptions are out of scope).
+
+## Classes involved
+
+### `ApolloClient`
+
+The main entry point for the runtime.
+
+Has a reference to an `HttpNetworkTransport`, as well as a list of `ApolloInterceptor`s.
+
+### `Operation` (interface)
+
+Represents a GraphQL operation (mutation, query or subscription).
+
+The generated classes for operations implement this interface.
+
+### `ApolloCall`
+
+Has a reference to an `ApolloClient` and an `Operation` and provides a method to make the `ApolloClient` execute
+the `Operation`.
```

Comment:
One key thing about `ApolloCall` is that it is mutable. It's one of the only classes (maybe the only?) mutable class that is not a `Builder`. It allows to chain configuration calls.